biogrid/libs/biogrid-simulator/src/lib/building/building.ts

```diff
@@ -14,11 +14,20 @@ export class Building implements EnergyUser {
    * @param {number} energy Amount of energy the building will have in joules.
    */
   constructor(energy: number) {
-    this.energyInJoules = energy;
+    if (this.isPositive(energy)) {
+      this.energyInJoules = energy;
+    } else {
+      throw new Error(`Can't create a building with negative energy!`);
+    }
+    
     // Initial id value, will be changed by rural area.
     this.buildingId = -1;
```

Comment:
If this value is going to be changed by the rural area, shouldn't there be a setter or maybe leave it to be public. The former is better.